test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -840,6 +840,302 @@ public void TestTextLoaderBackCompat_VerWritt_0x0001000C()
+            string dataPathDecimalMarkerPeriod = GetDataPath("iris.txt");
+            var readerDecimalMarkerComma = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.UInt32, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
+                        },
+                DecimalMarker = ','
+            });
+            var textDataMismatched1 = readerDecimalMarkerComma.Load(GetDataPath(dataPathDecimalMarkerPeriod));
+
+            // Check that the features being loaded are NaN.
+            DataViewSchema columnsPeriod = textDataMismatched1.Schema;
+            using DataViewRowCursor cursorPeriod = textDataMismatched1.GetRowCursor(columnsPeriod);
+            VBuffer<Single> featuresPeriod = default;
+            ValueGetter<VBuffer<Single>> featuresDelegatePeriod = cursorPeriod.GetGetter<VBuffer<Single>>(columnsPeriod[1]);
+            
+            // Iterate over each row
+            while (cursorPeriod.MoveNext())
+            {
+                featuresDelegatePeriod.Invoke(ref featuresPeriod);
+                foreach(float feature in featuresPeriod.GetValues().ToArray())
+                    Assert.Equal(feature, Single.NaN);
+            }
+
+            // Try reading a dataset where ',' is the actual decimal marker, but DecimalMarker = '.'.
+            string dataPathDecimalMarkerComma = GetDataPath("iris-decimal-marker-as-comma.txt");
+            var readerDecimalMarkerPeriod = new TextLoader(mlContext, new TextLoader.Options()
+            {
+                Columns = new[]
+                        {
+                            new TextLoader.Column("Label", DataKind.UInt32, 0),
+                            new TextLoader.Column("Features", DataKind.Single, new [] { new TextLoader.Range(1, 4) }),
+                        },
+                DecimalMarker = '.'
+            });
+            var textDataMismatched2 = readerDecimalMarkerPeriod.Load(GetDataPath(dataPathDecimalMarkerComma));
+
+            DataViewSchema columnsComma = textDataMismatched2.Schema;
+            using DataViewRowCursor cursorComma = textDataMismatched2.GetRowCursor(columnsComma);
+            VBuffer<Single> featuresComma = default;
+            ValueGetter<VBuffer<Single>> featuresDelegateComma = cursorComma.GetGetter<VBuffer<Single>>(columnsComma[1]);
+
+            // Iterate over each row
+            while (cursorComma.MoveNext())
+            {
+                featuresDelegateComma.Invoke(ref featuresComma);
+                foreach (float feature in featuresComma.GetValues().ToArray())
+                    Assert.Equal(feature, Single.NaN);
+            }
+        }
+
+        [Fact]
+        public void TestCommaAsDecimalMarkerWithSeperatorAsCommaInCSV()
+        {
```

Comment:
Is it possible to combine this test with TestCommaAsDecimalMarkerFloat? I think in both cases you're loading Single columns, from a baseline and from another file, and comparing them. So I think in this case using InlineData could be effective to branch between options.

Reply:
Good suggestion, I've made this change as well.